src/index.js

```diff
@@ -95,6 +95,8 @@ function sendPayment (params) {
 function executePayment (sourceTransfer, params) {
   return co(function * () {
     const isAtomic = !!params.notary
+    const isOptimistic = !params.receiptCondition && !isAtomic
```

Comment:
I think it would be better to have an explicit option for optimistic mode like `unsafeOptimisticTransport`. That way you can't accidentally shoot yourself in the foot if your condition happens to be null.